types/index.d.ts

```diff
@@ -0,0 +1,43 @@
+/// <reference types="typescript" />
+import { RequestInit, BodyInit } from 'node-fetch';
+
+// JSON types via https://github.com/microsoft/TypeScript/issues/1897#issuecomment-338650717
+type AnyJson = boolean | number | string | null | JsonArray | JsonMap;
+interface JsonMap {
+  [key: string]: AnyJson;
+}
```

Comment:
Just out of curiosity, could this also be written as:

```js
interface JsonMap extends Record<string, AnyJson> {}
```

If so, is one preferred over the other?

Reply:
Most likely*, or maybe even better as

```ts
type JsonMap = Record<string, AnyJson>;
```

*there might be an issue with the recursive types, but you will probably just have to try it out to see what works.

---

Overall I don't think one is preferred above the other.

Reply:
After testing it, `type JsonMap = Record<string, AnyJson>;` does throw an issue due to circularly referencing itself. However, `interface JsonMap extends Record<string, AnyJson> {}` does work without issue.